media-cdn/snippets.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+#
+# Copyright 2022 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to perform operations on data (content)
+when using Google Cloud Media CDN (Global Edge Network for Streaming Media).
+
+For more information, see the README.md under /media-cdn and the documentation
+at https://cloud.google.com/media-cdn/docs.
+"""
+
+import argparse
+import base64
+import datetime
+
+import cryptography.hazmat.primitives.asymmetric.ed25519 as ed25519
+
+
+from six.moves import urllib
+
+
+# [START mediacdn_sign_url]
+def sign_url(url, key_name, base64_key, expiration_time):
+    """Gets the Signed URL string for the specified URL and configuration.
+
+    Args:
+        url: URL to sign as a string.
+        key_name: name of the signing key as a string.
+        base64_key: signing key as a base64 encoded byte string.
+        expiration_time: expiration time as a UTC datetime object.
+
+    Returns:
+        Returns the Signed URL appended with the query parameters based on the
+        specified configuration.
+    """
+    stripped_url = url.strip()
+    parsed_url = urllib.parse.urlsplit(stripped_url)
+    query_params = urllib.parse.parse_qs(
+        parsed_url.query, keep_blank_values=True)
+    epoch = datetime.datetime.utcfromtimestamp(0)
+    expiration_timestamp = int((expiration_time - epoch).total_seconds())
+    decoded_key = base64.urlsafe_b64decode(base64_key)
+
+    url_pattern = u'{url}{separator}Expires={expires}&KeyName={key_name}'
+
+    url_to_sign = url_pattern.format(
+            url=stripped_url,
+            separator='&' if query_params else '?',
+            expires=expiration_timestamp,
+            key_name=key_name)
+
+    digest = ed25519.Ed25519PrivateKey.from_private_bytes(
+        decoded_key).sign(url_to_sign.encode('utf-8'))
+    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    signature = signature[:-2]  # to remove trailing '=' symbols
```

Comment:
You don't need to drop the trailing `=`.  Media CDN will handle the signatures either way.  (The `=` is just padding in base64.)

I'd drop this.

If you decide to keep it for other reasons, I would do `signature = signature.rstrip('=')` instead since base64 could have either 1 or 2 characters of padding.